Sources/SKCore/BuildSystemDelegate.swift

```diff
@@ -0,0 +1,23 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import Basic
+
+/// Handles  build system events, such as file build settings changes.
+public protocol BuildSystemDelegate: AnyObject {
+
+  /// Notify the delegate that the given files' build settings have changed.
+  ///
+  /// The callee should request new build settings for any of the given files that they are interested in.
+  func fileBuildSettingsChanged(_ changedFiles: Set<URL>)
```

Comment:
Even if we still have the `settings` method, I would like us to add the new settings here as discussed.  I think that makes this a `[URL: FileBuildSettings]`.

Reply:
This change is out of scope for now since it would require more LSP changes, once we have a proper back channel this should no longer be needed